src/components/shared/WithSize.js

```diff
@@ -36,23 +36,23 @@ export function withSize<
   $ReadOnly<$Diff<Props, SizeProps>>
 > {
   return class WithSizeWrapper extends React.PureComponent<*, State> {
-    _resizeListener: Event => void;
+    _resizeListener: () => void;
     state = { width: 0, height: 0 };
 
-    _observeSize = (wrappedComponent: React.Component<any> | null) => {
-      if (!wrappedComponent) {
-        return;
-      }
-      const container = findDOMNode(wrappedComponent); // eslint-disable-line react/no-find-dom-node
+    componentDidMount() {
+      const container = findDOMNode(this); // eslint-disable-line react/no-find-dom-node
       if (!container) {
         throw new Error('Unable to find the DOMNode');
       }
       this._resizeListener = () => {
         this._updateWidth(container);
       };
       window.addEventListener('resize', this._resizeListener);
-      this._updateWidth(container);
-    };
+
+      // Wrapping the first update in a requestAnimationFrame to defer the
+      // calculation until the full render is done.
+      requestAnimationFrame(() => this._updateWidth(container));
```

Comment:
Does `componentDidMount()` get called before the full render is done?

Reply:
`componentDidMount` is called synchronously after this element is appended to the DOM, but before any later sibling element are appended :(

So if the tree looks like this:

* Root 
  * Container
    * WithSize(Canvas)
    * Panels
  * Sidebar

When `componentDidMount` is called, the Sidebar isn't in the DOM yet (nor Panels, I think, but this wasn't a problem for our current code).

This is the proper function to add event listeners to the DOM element for _this_ container, but here we want to get the element's size _after_ all the tree is rendered, and that's why we need to delay it further.

Reply:
> componentDidMount is called synchronously after this element is appended to the DOM, but before any later sibling element are appended :(

Yikes! I wouldn't have expected that. All right then.